src/nvim/lua/treesitter.c

```diff
@@ -128,18 +128,18 @@ static const TSLanguage *load_language_from_object(lua_State *L, const char *pat
 {
   uv_lib_t lib;
   if (uv_dlopen(path, &lib)) {
-    uv_dlclose(&lib);
     luaL_error(L, "Failed to load parser for language '%s': uv_dlopen: %s",
                lang_name, uv_dlerror(&lib));
+    uv_dlclose(&lib);
```

Comment:
This is unreachable as `luaL_error` causes a long jump in C.

Reply:
We can't store `const char *` returned by `uv_dlerror`, it is not a static storage and can be freed after dlclose.
`strdup` does not work either for it may cause memory leak :exploding_head:

Reply:
Why can't we use a stack allocated string?

Reply:
"128 chars; that ought to be enough for anyone"

Reply:
> "128 chars; that ought to be enough for anyone"

In fact I just wrote `char [128]`, then I found something like
```
/usr/lib/libtree-sitter-markdown.so: cannot open shared object file: No such file or directory
```
took up almost 100 chars :rofl:

Reply:
Wait till you meet its brother, `libtree-sitter-markdown-inline.so`

Reply:
Give it 1MB for all I care. Errors are allowed to be slow.